imageregistry/v1/types_imagepruner.go

```diff
@@ -0,0 +1,100 @@
+package v1
+
+import (
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ImagePrunerList is a slice of ImagePruner objects.
+type ImagePrunerList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
+	Items           []ImagePruner `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ImagePruner is the configuration object for an image registry pruner
+// managed by the registry operator.
+type ImagePruner struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec ImagePrunerSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+	// +optional
+	Status ImagePrunerStatus `json:"status" protobuf:"bytes,3,opt,name=status"`
+}
+
+// ImagePrunerHistory is a configuration object for how many
+// successful and failed finished jobs to retain.
+type ImagePrunerHistory struct {
+	// SuccessfulJobsHistoryLimit specifies how many successful finished jobs to retain.
+	// Defaults to 3 if not set
+	// +optional
+	SuccessfulJobsHistoryLimit *int32 `json:"successfulJobsHistoryLimit,omitempty" protobuf:"bytes,1,opt,name=successfulJobsHistoryLimit"`
+	// FailedJobsHistoryLimit specifies how many failed finished jobs to retain.
+	// Defaults to 3 if not set
+	// +optional
+	FailedJobsHistoryLimit *int32 `json:"failedJobsHistoryLimit,omitempty" protobuf:"bytes,2,opt,name=failedJobsHistoryLimit"`
+}
+
+// ImagePrunerSpec defines the specs for the running image pruner.
+type ImagePrunerSpec struct {
+	// Schedule specifies when to execute the job
```

Comment:
it'll go to `oc explain`, so we should use json names there (i.e. it shouldn't start with a capital letter).

Reply:
also this line should end with the dot symbol. Otherwise we'll have `... when to execute the job Uses standard cronjob...`.

Reply:
added the period.
> also this line should end with the dot symbol. Otherwise we'll have `... when to execute the job Uses standard cronjob...`.

Reply:
> it'll go to `oc explain`, so we should use json names there (i.e. it shouldn't start with a capital letter).

updated in next push